lib/app_profiler/backend.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module AppProfiler
+  module Backend
+    class Base
```

Comment:
Let's move this to `lib/app_priofiler/backend/base.rb` and put autoloads for `Base`, `StackprofBackend`, `VernierBackend`. Also nitpick that the naming should either be `BaseBackend` and `<Name>Backend` as the subclass, or `Base` and `<Name>` as the subclass.